agent-control/src/sub_agent/on_host/command/command_os.rs

```diff
@@ -122,58 +137,56 @@ impl CommandOSStarted {
 
         Ok(self)
     }
-}
-
-#[cfg(target_family = "unix")]
-mod unix {
-    use tracing::warn;
-
-    use crate::sub_agent::on_host::command::{command_os::CommandOSStarted, error::CommandError};
-
-    use std::time::Duration;
-    const POLL_INTERVAL: Duration = Duration::from_millis(100);
-
-    impl CommandOSStarted {
-        pub fn shutdown(&mut self) -> Result<(), CommandError> {
-            let pid = self.get_pid() as i32;
 
-            use nix::{sys::signal, unistd::Pid};
-            let graceful_shutdown_result = signal::kill(Pid::from_raw(pid), signal::SIGTERM)
-                .inspect_err(|err| warn!(agent_id = %self.agent_id, "Failed to gracefully exit process {pid}: {err}. Attempting forceful shutdown"));
+    fn is_running_after_timeout(&mut self, timeout: Duration) -> bool {
+        let deadline = Instant::now() + timeout;
 
-            if graceful_shutdown_result.is_err()
-                || self.is_running_after_timeout(self.shutdown_timeout)
-            {
-                self.process.kill().map_err(CommandError::from)?;
+        while Instant::now() < deadline {
+            if self.is_running() {
+                std::thread::sleep(POLL_INTERVAL);
+            } else {
+                return false;
             }
-            Ok(())
         }
 
-        fn is_running_after_timeout(&mut self, timeout: Duration) -> bool {
-            let deadline = std::time::Instant::now() + timeout;
+        true
+    }
 
-            while std::time::Instant::now() < deadline {
-                if self.is_running() {
-                    std::thread::sleep(POLL_INTERVAL);
-                } else {
-                    return false;
-                }
-            }
+    pub fn shutdown(&mut self) -> Result<(), CommandError> {
+        let pid = self.get_pid();
+
+        // Attempt a graceful shutdown (platform-dependent).
+        #[cfg(not(any(target_family = "windows", target_family = "unix")))]
+        let graceful_shutdown_result =
+            Err(io::Error::other("Unsupported platform for graceful shutdown").into());
```

Comment:
Are we generally taking care of these targets? Would it make sense to use [compile_error](https://doc.rust-lang.org/std/macro.compile_error.html) so we assure we are actively taking care of it if ever supporting such targets?